lib/membrane_rtc_engine/endpoints/hls_endpoint.ex

```diff
@@ -298,7 +302,12 @@ if Enum.all?(
             hls_mode: state.hls_mode
           }
 
-          {hls_link_both_tracks(link, track, hls_sink_bin, state), state}
+          audio_encoding =
+            if track_type == :audio,
+              do: track.encoding,
+              else: get_audio_encoding(track.stream_id, state.tracks)
+
+          {hls_link_both_tracks(link, track, hls_sink_bin, audio_encoding, state), state}
```

Comment:
You can get `audio_encoding` always inside `hls_link_both_tracks` function as you pass both track and state.